contrib/inventory/ec2.py

```diff
@@ -789,11 +793,18 @@ def add_instance(self, instance, region):
             else:
                 hostname = getattr(instance, self.hostname_variable)
 
+        # set the hostname from route53
+        if self.route53_enabled and self.route53_hostnames:
+            route53_names = self.get_instance_route53_names(instance)
+            for name in route53_names:
+                if name.endswith(self.route53_hostnames):
+                    hostname = name
+
         # If we can't get a nice hostname, use the destination address
         if not hostname:
             hostname = dest
         else:
-            hostname = self.to_safe(hostname).lower()
+            hostname = hostname.lower()
```

Comment:
Okay, will add conditional.

My feeling was that the to_safe here is a bit of a misfeature in the first place, because people have already been working around it by adding changes like https://github.com/ansible/ansible/pull/15578 -- and over time you might wind up with a bunch of conditionals in this section, which could get ugly.

But I agree preserving backwards compatibility and not risking playbook breakage should be priority #1. Making change as suggested.